src/components/Button/Button.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+import { ReactComponent as PlusIcon } from 'assets/plus-icon.svg';
+import './Button.scss';
+
+const Button = React.forwardRef(({ type, icon, children, link, onClick, disabled }, ref) => {
+  const hasIcon = (icon) => icon ? <PlusIcon className='button__plus-icon'/> : null;
+
+  const rootURL = new RegExp('huoa.org/')
+  if (rootURL.test(link)) {
+    link = link.slice(rootURL.exec(link).index + 8)
+  }
+  
+  if (!onClick && !link) {
+    throw new Error('Button requires a link or an onClick attribute');
+  } 
+  else if (onClick) {
+    return (
+      <button className='button' type={type} ref={ref} onClick={onClick} disabled={disabled}>
+        {hasIcon(icon)}
```

Comment:
lets just make it `hasIcon()` no need to pass icon down since we have it currently is scope of the component

Reply:
good catch